llvm/lib/CodeGen/TargetLoweringBase.cpp

```diff
@@ -1607,6 +1607,10 @@ void TargetLoweringBase::computeRegisterProperties(
         // Try to widen the vector.
         for (unsigned nVT = i + 1; nVT <= MVT::LAST_VECTOR_VALUETYPE; ++nVT) {
           MVT SVT = (MVT::SimpleValueType) nVT;
+          // Skip RISCV vector tuple types since they don't involve in any
+          // widen/narrow operation.
+          if (SVT.isRISCVVectorTuple())
```

Comment:
Why is isRISCVVectorTuple in the VECTOR_VALUE_TYPE range?

Reply:
Because tuple type encode the `NF` in `nElem` and set `isVector = true`, so it's considered as a member in vector group.

Reply:
Does it need to set isVector = true? Can it be its own thing? The fewer places that need to special case it the better.

Reply:
Yeah, I can try to make it an independent thing rather than a special case of vector.